compiler/passes/flattenFunctions.cpp

```diff
@@ -376,8 +376,18 @@ addVarsToActuals(CallExpr* call, SymbolMap* vars, bool outerCall) {
   }
 }
 
-void flattenNestedFunctions(Vec<FnSymbol*>& nestedFunctions) {
-  compute_call_sites();
+static void deleteCalledby(FnSymbol* fn) {
+  if (fn->calledBy != NULL)  { delete fn->calledBy; fn->calledBy = NULL; }
+}
+static void deleteAllCalledby() {
+  for_alive_in_Vec(FnSymbol, fn, gFnSymbols)  deleteCalledby(fn);
+}
+
+void flattenNestedFunctions(Vec<FnSymbol*>& nestedFunctions, bool fastCCS) {
+  if (fastCCS)
+    { if (fVerify) deleteAllCalledby(); }
```

Comment:
Why only call `deleteAllCalledby` if in verify mode? And below only if not in verify mode?

Reply:
We need to delete `fn->calledBy`, done by deleteCalledby(fn), prior to calling compute_fn_call_sites(fn), because the latter would append to `calledBy` if it were present. We do not want that because `fn->calledBy` contains unknown contents prior to executing this code.

If `fVerify`, the code takes the more-expensive action of wiping out `calledBy` for **all** FnSymbols. This way, unintended `calledBy` access would result in a null pointer dereference, bringing that to our attention.